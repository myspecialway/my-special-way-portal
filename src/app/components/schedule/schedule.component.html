<table class="mat-table schedule-table" role="grid">
  <thead>
    <tr class="mat-header-row" role="row">
      <th class="mat-header-cell cdk-column-position mat-column-position" role="columnheader" colspan="2">שעה</th>
      <th class="mat-header-cell cdk-column-position mat-column-position" role="columnheader" *ngFor="let day of daysLabels">{{day}}</th>
    </tr>
  </thead>
  <tbody>
    <tr class="mat-row" role="row" *ngFor="let row of schedule; index as hourIndex">
      <td class="mat-cell cdk-column-position mat-column-position" role="gridcell">{{hourIndex + 1}}</td>
      <td class="mat-cell cdk-column-position mat-column-position" role="gridcell">{{hoursLabels[hourIndex]}}</td>
      <td 
        class="mat-cell cdk-column-position mat-column-position"
        *ngFor="let timeslot of row; index as dayIndex" 
        role="gridcell" 
        (click)="timeSlotClicked.emit({hourIndex: hourIndex, dayIndex: dayIndex})"
      >
        <div *ngIf="timeslot.lesson" class="ts-full">
          <span class="ts-lesson">{{timeslot.lesson.title}}</span>
          <span class="ts-location" *ngIf="timeslot.location">{{timeslot.location.name}}</span>
        </div>
        <div *ngIf="!timeslot.lesson" class="ts-empty">
          <i class="material-icons" >
            add
          </i>
        </div>
      </td>
    </tr>
  </tbody>
</table>
