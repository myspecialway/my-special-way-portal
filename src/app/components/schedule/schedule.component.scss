$hover: rgba(106, 0, 207, 0.2);

.schedule-table {
    width: 100%;
    direction: rtl;
    font-family: "Rubik", "Helvetica", "Arial", sans-serif;
    color: #222;
    th.mat-header-cell {
        background-color: #fafafa;
        font-size: 16px;
        font-weight: bold;
        color: #222;
        border: 0;
    }
    td.mat-cell, td.mat-footer-cell, th.mat-header-cell {
        text-align: right;
        padding: 20px 14px;
    }
    td.mat-cell, td.mat-footer-cell {
        cursor: pointer;
        border: 1px solid #e0e0e0;
        font-size: 18px;
        &:hover {
            background-color: $hover;
        }
    }

    td.mat-cell:first-child, td:nth-child(2) {
        text-align: center;
        direction: initial;
        font-weight: 400;
    }
    
    .ts-lesson, .ts-location {
        font-family: Rubik;
        color: #222;
        display: block;
    }
    .ts-lesson {
        font-size: 18px;
        font-weight: 400;
    }
    .ts-location {
        font-size: 14px;
    }
    .ts-empty {
        text-align: center;
        color: #6a00cf;
        opacity: .5;
        &:hover {
            opacity: 1;
        }
    }
}
