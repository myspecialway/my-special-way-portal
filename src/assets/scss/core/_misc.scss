body {
  background-color: #fafafa;
  color: $black-color;
  font-weight: 300;
}

legend {
  border-bottom: 0;
}

.serif-font {
  font-family: $font-family-serif;
}

// Prevent highlight on mobile
* {
  -webkit-tap-highlight-color: rgba(255, 255, 255, 0);
  -webkit-tap-highlight-color: transparent;
  &:focus {
    outline: 0;
  }
}

a {
  color: $link-color;
  &:hover,
  &:focus {
    color: darken($link-color, 5%);
    text-decoration: none;
  }

  &.text-info {
    &:hover,
    &:focus {
      color: darken($brand-info, 5%);
    }
  }

  & .material-icons {
    vertical-align: middle;
  }
}

.form-check,
label {
  font-size: 14px;
  line-height: 1.42857;
  color: $checkboxes-text-color;
  font-weight: 400;
}

/*           Animations              */
.animation-transition-general {
  @include transition-all($general-transition-time, $transition-linear);
}

.animation-transition-slow {
  @include transition-all($slow-transition-time, $transition-linear);
}

.animation-transition-fast {
  @include transition-all($fast-transition-time, $transition-ease);
}

.caret,
.sidebar a {
  @include transition-all($fast-transition-time, $transition-ease-in);
}
#map {
  position: relative;
  z-index: 2;
  width: 100%;
  height: calc(100vh - 70px);
  margin-top: 70px;
}

@font-face {
  font-family: 'pass';
  font-style: normal;
  font-weight: 400;
  src: url(data:application/font-woff;charset=utf-8;base64,d09GRgABAAAAAATsAA8AAAAAB2QAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAABGRlRNAAABWAAAABwAAAAcg9+z70dERUYAAAF0AAAAHAAAAB4AJwANT1MvMgAAAZAAAAA/AAAAYH7AkBhjbWFwAAAB0AAAAFkAAAFqZowMx2N2dCAAAAIsAAAABAAAAAQAIgKIZ2FzcAAAAjAAAAAIAAAACAAAABBnbHlmAAACOAAAALkAAAE0MwNYJ2hlYWQAAAL0AAAAMAAAADYPA2KgaGhlYQAAAyQAAAAeAAAAJAU+ATJobXR4AAADRAAAABwAAAAcCPoA6mxvY2EAAANgAAAAEAAAABAA5gFMbWF4cAAAA3AAAAAaAAAAIAAKAE9uYW1lAAADjAAAARYAAAIgB4hZ03Bvc3QAAASkAAAAPgAAAE5Ojr8ld2ViZgAABOQAAAAGAAAABuK7WtIAAAABAAAAANXulPUAAAAA1viLwQAAAADW+JM4eNpjYGRgYOABYjEgZmJgBEI2IGYB8xgAA+AANXjaY2BifMg4gYGVgYVBAwOeYEAFjMgcp8yiFAYHBl7VP8wx/94wpDDHMIoo2DP8B8kx2TLHACkFBkYA8/IL3QB42mNgYGBmgGAZBkYGEEgB8hjBfBYGDyDNx8DBwMTABmTxMigoKKmeV/3z/z9YJTKf8f/X/4/vP7pldosLag4SYATqhgkyMgEJJnQFECcMOGChndEAfOwRuAAAAAAiAogAAQAB//8AD3jaY2BiUGJgYDRiWsXAzMDOoLeRkUHfZhM7C8Nbo41srHdsNjEzAZkMG5lBwqwg4U3sbIx/bDYxgsSNBRUF1Y0FlZUYBd6dOcO06m+YElMa0DiGJIZUxjuM9xjkGRhU2djZlJXU1UDQ1MTcDASNjcTFQFBUBGjYEkkVMJCU4gcCKRTeHCk+fn4+KSllsJiUJEhMUgrMUQbZk8bgz/iA8SRR9qzAY087FjEYD2QPDDAzMFgyAwC39TCRAAAAeNpjYGRgYADid/fqneL5bb4yyLMwgMC1H90HIfRkCxDN+IBpFZDiYGAC8QBbSwuceNpjYGRgYI7594aBgcmOAQgYHzAwMqACdgBbWQN0AAABdgAiAAAAAAAAAAABFAAAAj4AYgI+AGYB9AAAAAAAKgAqACoAKgBeAJIAmnjaY2BkYGBgZ1BgYGIAAUYGBNADEQAFQQBaAAB42o2PwUrDQBCGvzVV9GAQDx485exBY1CU3PQgVgIFI9prlVqDwcZNC/oSPoKP4HNUfQLfxYN/NytCe5GwO9/88+/MBAh5I8C0VoAtnYYNa8oaXpAn9RxIP/XcIqLreZENnjwvyfPieVVdXj2H7DHxPJH/2/M7sVn3/MGyOfb8SWjOGv4K2DRdctpkmtqhos+D6ISh4kiUUXDj1Fr3Bc/Oc0vPqec6A8aUyu1cdTaPZvyXyqz6Fm5axC7bxHOv/r/dnbSRXCk7+mpVrOqVtFqdp3NKxaHUgeod9cm40rtrzfrt2OyQa8fppCO9tk7d1x0rpiQcuDuRkjjtkHt16ctbuf/radZY52/PnEcphXpZOcofiEZNcQAAeNpjYGIAg///GBgZsAF2BgZGJkZmBmaGdkYWRla29JzKggxD9tK8TAMDAxc2D0MLU2NjENfI1M0ZACUXCrsAAAABWtLiugAA)
    format('woff');
}

input[type='password'] {
  font-family: 'pass', 'Roboto', Helvetica, Arial, sans-serif;
  font-size: 18px;
  &::-webkit-input-placeholder {
    transform: scale(0.77);
    transform-origin: 0 50%;
  }
  &::-moz-placeholder {
    font-size: 14px;
    opacity: 1;
  }
  &:-ms-input-placeholder {
    font-size: 14px;
    font-family: 'Roboto', Helvetica, Arial, sans-serif;
  }
}
