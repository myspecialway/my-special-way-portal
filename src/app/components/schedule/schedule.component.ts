import { Component, Input, Output, EventEmitter } from '@angular/core';
import { TimeSlot } from '../../models/timeslot.model';

export interface TimeSlotIndexes {
  hourIndex: number;
  dayIndex: number;
}

@Component({
  selector: 'app-schedule',
  templateUrl: './schedule.component.html',
  styleUrls: ['./schedule.component.scss'],
})

export class ScheduleComponent {

  @Input()
  schedule: TimeSlot[][];
  @Input()
  daysLabels: string[];
  @Input()
  hoursLabels: string[];
  @Output()
  timeSlotClicked: EventEmitter<TimeSlotIndexes> = new EventEmitter();
}
