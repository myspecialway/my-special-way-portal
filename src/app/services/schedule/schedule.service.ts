import { Injectable } from '@angular/core';

@Injectable()
export class ScheduleService {
  levels = ['א', 'ב', 'ג', 'ד', 'ה', 'ו'];
  daysLabels = ['ראשון', 'שני', 'שלישי', 'רביעי', 'חמישי', 'שישי'];
  hoursLabels = [
    '07:30 - 08:00',
    '08:00 - 08:50',
    '08:50 - 09:35',
    '09:35 - 10:00',
    '10:00 - 10:30',
    '10:30 - 11:15',
    '11:15 - 12:00',
    '12:00 - 12:45',
    '12:45 - 13:15',
    '13:15 - 13:45',
    '13:45 - 14:30',
    '14:30 - 15:15',
    '15:15 - 16:00',
    '16:00 - 16:30',
    '16:30 - 16:45',
  ];
}
