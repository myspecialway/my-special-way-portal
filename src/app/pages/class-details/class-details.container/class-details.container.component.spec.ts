import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ClassService } from '../../class/services/class.graphql.service';
import { ScheduleService } from '../../../services/schedule/schedule.service';
import { ActivatedRoute } from '@angular/router';
import { ClassDetailsContainerComponent } from './class-details.container.component';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { MatDialog } from '@angular/material';
import { Observable } from 'rxjs-compat';
import { ScheduleDialogData } from '../../../components/schedule/schedule-dialog/schedule-dialog-data.model';

describe('ClassDetailsContainerComponent', () => {
  let component: ClassDetailsContainerComponent;
  let classServiceMock: Partial<ClassService>;
  let scheduleServiceMock: Partial<ScheduleService>;
  let scheduleDialogMock: Partial<MatDialog>;
  let fixture: ComponentFixture<ClassDetailsContainerComponent>;
  let afterClosedMockFn: jest.Mock;
  const mockedScheduleDialogData = {
    index: '00',
    lesson: {
      _id: '5b2abc74572e7619a628c11c',
      title: 'test lesson',
      icon: 'test-icon',
    },
    location: {
      _id: '5b5596a7739a882933edd4fc',
      disabled: false,
      name: 'test location',
      position: {
        latitude: 0,
        longitude: 0,
        floor: 1,
      },
    },
  } as ScheduleDialogData;

  const mockedClass = {
    data: {
      classById: {
        _id: '5b217b030825622c97d3757f',
        level: 'א',
        number: 1,
        name: 'טיטאן',
        schedule: [
          {
            index: '10',
            lesson: {
              _id: '5b2abc74572e7619a628c11c',
              title: 'test lesson',
              icon: 'test-icon',
            },
            location: {
              _id: '5b5596a7739a882933edd4fc',
              disabled: false,
              name: 'test location',
              position: {
                latitude: 0,
                longitude: 0,
                floor: 1,
              },
            },
          },
        ],
      },
    },
  };
  beforeEach(async () => {
    afterClosedMockFn = jest.fn().mockReturnValue(
      Observable.of(mockedScheduleDialogData),
    );
    classServiceMock = {
      classById: jest.fn().mockReturnValue(Promise.resolve(mockedClass)),
      update: jest.fn().mockReturnValue(Promise.resolve({data: {updateClass: {_id: 'updateclassid'}}})),
    };
    scheduleServiceMock = {
      levels: ['א', 'ב', 'ג'],
      daysLabels: ['ראשון', 'שני', 'שלישי', 'רביעי', 'חמישי', 'שישי'],
      hoursLabels: [
        '07:30 - 08:00',
        '08:00 - 08:50',
        '08:50 - 09:35',
      ],
    },
    scheduleDialogMock = {
      open: jest.fn().mockReturnValue({
        afterClosed: afterClosedMockFn,
      }),
    };
    TestBed.configureTestingModule({
      declarations: [ClassDetailsContainerComponent],
      providers: [
        { provide: ClassService, useValue: classServiceMock },
        {
          provide: ActivatedRoute,
          useValue: {
            snapshot: { params: { id: '5b217b030825622c97d3757f' } },
          },
        },
        { provide: MatDialog, useValue: scheduleDialogMock },
        { provide: ScheduleService, useValue: scheduleServiceMock },
      ],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ClassDetailsContainerComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toMatchSnapshot();
  });

  it('should open a dialog when clicking on an empty timeslot', () => {
    fixture.componentInstance.onTimeSlotClick({hourIndex: 0, dayIndex: 0});
    const DialogMock = TestBed.get(MatDialog);
    expect(DialogMock.open).toHaveBeenCalled();
  });

  it('should open a dialog when clicking on an existing timeslot', () => {
    afterClosedMockFn.mockReset();
    afterClosedMockFn.mockReturnValueOnce(Observable.of({...mockedScheduleDialogData, index: '10'}));
    fixture.componentInstance.onTimeSlotClick({hourIndex: 1, dayIndex: 0});
    const DialogMock = TestBed.get(MatDialog);
    expect(DialogMock.open).toHaveBeenCalled();
  });

  it('should not update the class when dismissing the dialog', () => {
    afterClosedMockFn.mockReset();
    afterClosedMockFn.mockReturnValueOnce(Observable.of(undefined));
    fixture.componentInstance.onTimeSlotClick({hourIndex: 1, dayIndex: 0});
    expect(classServiceMock.update).not.toHaveBeenCalled();
  });

  it('should call onDetailChange', () => {
    fixture.componentInstance.onDetailChange({name: 'newName', level: 'newlevel'});
    const updatedClass = {...mockedClass.data.classById, name: 'newName', level: 'newlevel' };
    expect(classServiceMock.update).toHaveBeenCalledWith(updatedClass);
  });
});
