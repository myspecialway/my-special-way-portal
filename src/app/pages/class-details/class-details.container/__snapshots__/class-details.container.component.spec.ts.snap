// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassDetailsContainerComponent should create 1`] = `
ClassDetailsContainerComponent {
  "_class": Object {
    "_id": "5b217b030825622c97d3757f",
    "level": "א",
    "name": "טיטאן",
    "number": 1,
    "schedule": Array [
      Object {
        "index": "10",
        "lesson": Object {
          "_id": "5b2abc74572e7619a628c11c",
          "icon": "test-icon",
          "title": "test lesson",
        },
        "location": Object {
          "_id": "5b5596a7739a882933edd4fc",
          "disabled": false,
          "name": "test location",
          "position": Object {
            "floor": 1,
            "latitude": 0,
            "longitude": 0,
          },
        },
      },
    ],
  },
  "classService": Object {
    "classById": [MockFunction] {
      "calls": Array [
        Array [
          "5b217b030825622c97d3757f",
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": ZoneAwarePromise {
            "__zone_symbol__state": true,
            "__zone_symbol__value": Object {
              "data": Object {
                "classById": Object {
                  "_id": "5b217b030825622c97d3757f",
                  "level": "א",
                  "name": "טיטאן",
                  "number": 1,
                  "schedule": Array [
                    Object {
                      "index": "10",
                      "lesson": Object {
                        "_id": "5b2abc74572e7619a628c11c",
                        "icon": "test-icon",
                        "title": "test lesson",
                      },
                      "location": Object {
                        "_id": "5b5596a7739a882933edd4fc",
                        "disabled": false,
                        "name": "test location",
                        "position": Object {
                          "floor": 1,
                          "latitude": 0,
                          "longitude": 0,
                        },
                      },
                    },
                  ],
                },
              },
            },
          },
        },
      ],
    },
    "update": [MockFunction],
  },
  "dialog": Object {
    "open": [MockFunction],
  },
  "route": Object {
    "snapshot": Object {
      "params": Object {
        "id": "5b217b030825622c97d3757f",
      },
    },
  },
  "schedule": Array [
    Array [
      Object {
        "index": "00",
      },
      Object {
        "index": "01",
      },
      Object {
        "index": "02",
      },
      Object {
        "index": "03",
      },
      Object {
        "index": "04",
      },
      Object {
        "index": "05",
      },
    ],
    Array [
      Object {
        "index": "10",
        "lesson": Object {
          "_id": "5b2abc74572e7619a628c11c",
          "icon": "test-icon",
          "title": "test lesson",
        },
        "location": Object {
          "_id": "5b5596a7739a882933edd4fc",
          "disabled": false,
          "name": "test location",
          "position": Object {
            "floor": 1,
            "latitude": 0,
            "longitude": 0,
          },
        },
      },
      Object {
        "index": "11",
      },
      Object {
        "index": "12",
      },
      Object {
        "index": "13",
      },
      Object {
        "index": "14",
      },
      Object {
        "index": "15",
      },
    ],
    Array [
      Object {
        "index": "20",
      },
      Object {
        "index": "21",
      },
      Object {
        "index": "22",
      },
      Object {
        "index": "23",
      },
      Object {
        "index": "24",
      },
      Object {
        "index": "25",
      },
    ],
  ],
  "scheduleService": Object {
    "daysLabels": Array [
      "ראשון",
      "שני",
      "שלישי",
      "רביעי",
      "חמישי",
      "שישי",
    ],
    "hoursLabels": Array [
      "07:30 - 08:00",
      "08:00 - 08:50",
      "08:50 - 09:35",
    ],
    "levels": Array [
      "א",
      "ב",
      "ג",
    ],
  },
}
`;
